logic/AbstractCommonModel.cpp

```diff
@@ -0,0 +1,102 @@
+// Licensed under the Apache-2.0 license. See README.md for details.
+
+#include "AbstractCommonModel.h"
+
```

Comment:
Not having to implement a QAbstractItemModel everywhere. It is currently only used for AccountTypesModel (AccountModel.cpp#18-27), but if you take a look at it and compare to what you'd need to do without it you'll see the use of it. This is roughly what would be needed otherwise:

```cpp
class AccountTypesModel : public QAbstractListModel
{
    Q_OBJECT
public:
    explicit AccountTypesModel(QObject *parent = nullptr) : QAbstractItemModel(parent) {}

    int rowCount() const override
    {
        return m_rows.size();
    }
    QVariant data(const QModelIndex &index, const int role) const override
    {
        BaseAccountType *row = m_rows.at(index.row());
        switch (role)
        {
        case Qt::DisplayRole:
            return row->text();
        case Qt::UserRole:
            return row->id();
        case Qt::DecorationRole:
            return row->icon();
        }
        return QVariant();
    }

private:
    QList<BaseAccountType *> m_rows;
};
```
AccountTypesModel is even relatively simple. Add headers, multiple columns etc. and you have way more code. Also should be noted that the above does not have insertion/deletion, adding that would further increase the boilerplate code.

Having this will make more sense when used in more places, but I wanted to keep the changes low.